plugins/obs-outputs/rtmp-stream.c

```diff
@@ -940,13 +940,46 @@ static void *connect_thread(void *data)
 static bool rtmp_stream_start(void *data)
 {
 	struct rtmp_stream *stream = data;
+	obs_data_t *settings;
+
+	settings = obs_output_get_settings(stream->output);
 
 	if (!obs_output_can_begin_data_capture(stream->output, 0))
 		return false;
 	if (!obs_output_initialize_encoders(stream->output, 0))
 		return false;
 
 	os_atomic_set_bool(&stream->connecting, true);
+
+	/* store initial bitrate for dynamical variable bitrate option */
+	obs_encoder_t *vencoder = obs_output_get_video_encoder(stream->output);
+	const char *encoder_id = obs_encoder_get_id(vencoder);
+	obs_data_t *params = obs_encoder_get_settings(vencoder);
+	if (params) {
+		int bitrate = obs_data_get_int(params, "bitrate");
+		stream->initial_bitrate = bitrate;
+		stream->dynamic_bitrate = bitrate;
+		bool isSimpleMode = obs_data_get_bool(settings, "IsSimpleMode1") ||
+				obs_data_get_bool(settings, "IsSimpleMode2");
+		blog(LOG_INFO, "the mode is %i \n", isSimpleMode);
```

Comment:
What is the purpose of this log line?

Speaking of which, I see that there is a log entry when bitrate is adjusted, but is there an entry anywhere to indicate whether the user has enabled this feature at all?

Reply:
oh i just forgot to remove the line which helped me during dev.   
It was to track the bool. It is used for UI . Initially I had coded only for x264 but if other h264 encoders are supported I needed the optio nto appear in Simple and Advanced mode as well.   
I'll log the activation of variable bitrate, good idea.